base/namedtuple.jl

```diff
@@ -0,0 +1,213 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    NamedTuple{names,T}(args::Tuple)
+
+Construct a named tuple with the given `names` (a tuple of Symbols) and field types `T`
+(a `Tuple` type) from a tuple of values.
+"""
+function NamedTuple{names,T}(args::Tuple) where {names, T <: Tuple}
+    if length(args) == length(names)
+        if @generated
+            N = length(names)
+            types = T.parameters
+            Expr(:new, :(NamedTuple{names,T}), Any[ :(convert($(types[i]), args[$i])) for i in 1:N ]...)
+        else
+            N = length(names)
+            NT = NamedTuple{names,T}
+            ccall(:jl_new_structv, Any, (Any, Ptr{Void}, UInt32), NT, collect(Any, T(args)), N)::NT
```

Comment:
Are we sure that `collect(Any, T(args))` === `Any[ convert(T.parameters[i], arg[i]) for i in 1:N ]`?

Reply:
I think so. Is there a counter-example?

Reply:
Probably not currently. Just that, as written, we cannot use this code in Core.Inference (but again, currently we don't)

Reply:
Can we also add a test for these lines of code?

Reply:
Any ideas for convenient ways to test both implementations of generated functions?

Reply:
One idea: I could add a `call_unspecialized` function.